src/components/custom-field-input-text/custom-field-input-text.jsx

```diff
@@ -0,0 +1,64 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import cautionSvg from '../../svgs/icon-caution-fill.svg';
+import Icon from '../icon/icon.jsx';
+import styles from './custom-field-input-text.css';
+
+function getRootClassName(className, error, disabled) {
+  if (disabled) {
+    return `${className} ${styles.disabled}`;
+  }
+
+  if (error) {
+    return `${className} ${styles.error}`;
+  }
+
+  return className;
+}
+
+export default function CustomFieldInputText(props) {
+  return (
+    <div className={getRootClassName(props.className, props.error, props.disabled)}>
+      <div className={styles['heading-container']}>
+        <label className={styles.label} htmlFor={props.id}>Input Descriptor</label>
+        {props.required && <span className={styles.optional}>(Required)</span>}
+      </div>
+      <div className={styles['input-container']}>
+        <input type="text" id={props.id} name={props.name} placeholder={props.placeholder} value={props.value} onClick={props.onClick} disabled={props.disabled} />
+        {props.error &&
+          <div className={styles['input-icon-container']}>
+            <Icon className={styles['input-icon']} currentColor="caution" name={cautionSvg.id} size="medium" />
+          </div>
+        }
+      </div>
+      <span className={styles.help}>{props.helpText}</span>
+    </div>
+  );
+}
+
+CustomFieldInputText.propTypes = {
+  className: PropTypes.string,
+  disabled: PropTypes.bool,
+  error: PropTypes.bool,
+  helpText: PropTypes.string,
+  id: PropTypes.string,
+  name: PropTypes.string,
+  onClick: PropTypes.func,
+  placeholder: PropTypes.string,
+  required: PropTypes.bool,
+  value: PropTypes.string,
+};
+
+CustomFieldInputText.defaultProps = {
+  className: styles['custom-field-input-text'],
```

Comment:
Oh, this is another question I wanted to ask but forgot:

What are your thoughts around using `styles['custom-field-input-text']` versus `styles.customFieldInputText`? 

For context, we haven't really decided on either. As an engineering org, we have historically used the hyphenated case. But the recent introduction for CSS modules has had a few people wondering whether we should switch.

I'm indifferent TBH. But if we do develop some reasoning for either then we could probably add a linter rule to ensure consistency in our style sheet files.

Reply:
I am also indifferent as to which we choose, there are plenty of arguments for both... but I do think that we should choose, and implement linting rules, so... I'm very helpful, I know. 😄

Reply:
Well, let's just keep it in mind in case the engineering org decides to move in a direction or another... or both! Why not both...